tests/tests_sound_composer/test_sound_composer_source_harmonics.py

```diff
@@ -0,0 +1,490 @@
+    )
+    with pytest.raises(
+        PyAnsysSoundException,
+        match="Harmonics source control is not set. Use ``SourceHarmonics.source_control``.",
+    ):
+        source_harmonics_obj.process()
+
+    # Test process method exception2 (missing harmonics source data).
+    source_harmonics_obj.source_harmonics = None
+    f_source_control = fields_factory.create_scalar_field(
+        num_entities=1, location=locations.time_freq
+    )
+    f_source_control.append([500, 2000, 3000, 3500], 1)
+    source_control_obj = SourceControlTime()
+    source_control_obj.control = f_source_control
+    source_harmonics_obj.source_control = source_control_obj
+    with pytest.raises(
+        PyAnsysSoundException,
+        match=(
+            "Harmonics source data is not set. Use ``SourceHarmonics.source_harmonics`` "
+            "or method ``SourceHarmonics.load_source_harmonics\\(\\)``."
+        ),
+    ):
+        source_harmonics_obj.process()
+
+    # Test process method exception3 (invalid sampling frequency value).
+    source_harmonics_obj.load_source_harmonics(
+        pytest.data_path_sound_composer_harmonics_source_in_container
+    )
+    with pytest.raises(
+        PyAnsysSoundException, match="Sampling frequency must be strictly positive."
+    ):
+        source_harmonics_obj.process(sampling_frequency=0.0)
+
+
+def test_source_harmonics_get_output(dpf_sound_test_server):
+    """Test SourceHarmonics get_output method."""
+    # Create a field to use in a SourceControlTime object.
+    f_source_control = fields_factory.create_scalar_field(
+        num_entities=1, location=locations.time_freq
+    )
+    f_source_control.append([500, 2000, 3000, 3500], 1)
+    support = TimeFreqSupport()
+    f_time = fields_factory.create_scalar_field(num_entities=1, location=locations.time_freq)
+    f_time.append([0, 1, 2, 3], 1)
+    support.time_frequencies = f_time
+    f_source_control.time_freq_support = support
+
+    # Create a SourceControlTime object.
+    source_control_obj = SourceControlTime()
+    source_control_obj.control = f_source_control
+
+    source_harmonics_obj = SourceHarmonics(
+        pytest.data_path_sound_composer_harmonics_source_in_container,
+        source_control_obj,
+    )
+    source_harmonics_obj.process(sampling_frequency=44100.0)
+    f_output = source_harmonics_obj.get_output()
+    assert isinstance(f_output, Field)
+    assert len(f_output.data) / 44100.0 == pytest.approx(3.0, abs=1e-2)
+
+    # Compute the power spectral density over the output signal.
+    psd = PowerSpectralDensity(
+        input_signal=f_output,
+        fft_size=8192,
+        window_type="HANN",
+        window_length=8192,
+        overlap=0.75,
+    )
+    psd.process()
+    psd_squared, psd_freq = psd.get_PSD_squared_linear_as_nparray()
+    delat_f = psd_freq[1] - psd_freq[0]
+
+    # Check the sound power level in the octave bands centered at 250, 1000 and 4000 Hz.
```

Comment:
@ansaminard FYI in the case of Harmonics generation, the non-deterministic elements are only the initial phases of the generated sinus, which should not impact that much the PSD level of the generated signal. Tolerance may probably set to a lower value. (but anyway, let's keep it as is)

Reply:
@anshlachamb Yes, that's why I reduced it from 3 to 1 dB compared to BBN sources. I did a few trials before deciding on this limit values.